### PR TITLE
Add module sources

### PR DESCRIPTION
**Pull request by marekrozmus**

### CODE REVIEW

File: src/SwipeableList.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    super(props);
+
+    this.dragging = false;
+
+    // on list scrolling we block items swipe functionality
+    this.state = { blockSwipe: false };
+  }
+
+  componentDidMount() {
+    window.addEventListener('mouseup', this.onDragEndMouse);
```
Comment (przemyslawzalewski) [inline src/SwipeableList.js]: This would be a lot easier with hooks.

File: src/SwipeableList.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  onDragEnd = () => {
+    this.dragging = false;
+    this.setState({ blockSwipe: false });
+  };
+
+  render() {
+    const { children } = this.props;
+    const { blockSwipe } = this.state;
+
+    const childrenWithProps = React.Children.map(children, child =>
```
Comment (przemyslawzalewski) [inline src/SwipeableList.js]: Consider [render prop pattern](https://reactjs.org/docs/render-props.html) and or custom hook. `React.cloneElement` should never have existed IMHO.

File: src/SwipeableList.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        onScroll={this.onScroll}
+      >
+        {childrenWithProps}
+      </div>
+    );
+  }
+}
+
+SwipeableList.propTypes = {
+  children: PropTypes.node.isRequired
```
Comment (przemyslawzalewski) [inline src/SwipeableList.js]: What about string, number, etc. elements that can be rendered by React?

File: src/SwipeableListItem.js [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+    }
+  };
+
+  render() {
+    const { children, swipeLeft, swipeRight } = this.props;
+
+    return (
+      <div className="swipeable-list-item" ref={div => (this.wrapper = div)}>
+        {swipeLeft && (
+          <div ref={div => (this.backgroundLeft = div)} className="background">
```
Comment (przemyslawzalewski) [inline src/SwipeableListItem.js]: Please use cached function instances as refs as this causes an infinite loop of null, element, null, element... chains.

File: src/SwipeableListItem.js [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+        </div>
+      </div>
+    );
+  }
+}
+
+SwipeableListItem.propTypes = {
+  blockSwipe: PropTypes.bool,
+  children: PropTypes.node.isRequired,
+  swipeLeft: PropTypes.shape({
```
Comment (przemyslawzalewski) [inline src/SwipeableListItem.js]: This can be extracted as a `swipeActionPropType`.

File: src/index.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import SwipeableList from './SwipeableList';
+import SwipeableListItem from './SwipeableListItem';
+
+export {
```
Comment (przemyslawzalewski) [inline src/index.js]: Consider `export { default: SwipeableList, Item: SwipeableListItem }`

File: src/SwipeableListItem.js [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+  onTouchMove = evt => {
+    const touch = evt.targetTouches[0];
+    const delta = touch.clientX - this.dragStartX;
+
+    if (this.shouldMoveItem(delta)) {
+      this.left = delta;
+    }
+  };
+
+  updatePosition = () => {
```
Comment (przemyslawzalewski) [inline src/SwipeableListItem.js]: Consider the react-way instead of direct DOM manipulation unless necessary.

File: src/SwipeableListItem.js [hunk 8]
```diff
@@ -0,0 +1,236 @@
...
+
+  onMouseMove = ({ clientX }) => {
+    const delta = clientX - this.dragStartX;
+
+    if (this.shouldMoveItem(delta)) {
+      this.left = delta;
+    }
+  };
+
+  onTouchMove = evt => {
```
Comment (przemyslawzalewski) [inline src/SwipeableListItem.js]: Please use full names, "event" in this case. If the name is not needed, consider inline destructuring.

File: src/SwipeableList.js (hunk 3 above)
Comment (marekrozmus) [reply src/SwipeableList.js]: From documentation:
  // Anything that can be rendered: numbers, strings, elements or an array
  // (or fragment) containing these types.

File: .babelrc [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+        "targets": {
+          "browsers": ["last 2 versions", "safari >= 7"]
+        },
+        "modules": false
+      }
+    ],
+    "@babel/preset-react"
+  ],
+  "plugins": [
+    ["@babel/plugin-proposal-class-properties"]
```
[the remark this answers is not in the document]
Comment (marekrozmus) [reply .babelrc]: It is needed for arrow functions in classes. If not added then during build you'll get error: "Support for the experimental syntax 'classProperties' isn't currently enabled"

**Pull request merged**